app/models/supplejack_api/concerns/privacyable.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module SupplejackApi
+  module Concerns
+    module Privacyable
+      extend ActiveSupport::Concern
+
+      included do
+        field :privacy, type: String,   default: 'public'
+
+        validates :privacy, inclusion: { in: %w[public hidden private] }
+        before_validation :set_default_privacy
+
+        # can't add this scope without breaking the model
+        # it must be an existing method in Mongoid
+        # scope :public,              -> { publicly_viewable }
+        scope :publicly_viewable,     -> { where(privacy: 'public') }
+        scope :hidden,                -> { where(privacy: 'hidden') }
+        scope :private,               -> { where(privacy: 'private') }
+        scope :public_or_hidden,      -> { self.in(privacy: %w[public hidden]) }
+        scope :public_not_favourites, -> { where(privacy: 'public', :name.ne => 'Favourites') }
```

Comment:
I haven't seen the `:name.ne` before, that's pretty slick! Good bye all the chaining of `where`s into `where.not`s!

Reply:
Not sure but I believe that's mongoid specific.